noether_gui/src/widgets/tpp_widget.cpp

```diff
@@ -208,6 +234,62 @@ vtkSmartPointer<vtkAssembly> createToolPathActors(const std::vector<ToolPaths>&
   return assembly;
 }
 
+/*
+ *  Create a polyline between all points of the tool path
+ *
+ *  @param tool_paths The tool paths to create the polyline from
+ *  @param waypoint_shape_output_port The output port of the waypoint shape
+ *  @return The assembly containing the polyline
+ */
+vtkSmartPointer<vtkAssembly> createToolPathPolylineActor(const std::vector<ToolPaths>& tool_paths,
+                                                         vtkAlgorithmOutput* waypoint_shape_output_port)
+{
+  auto assembly = vtkSmartPointer<vtkAssembly>::New();
+  auto polyLine = vtkSmartPointer<vtkPolyLine>::New();
+  auto points = vtkSmartPointer<vtkPoints>::New();
+
+  for (const ToolPaths& fragment : tool_paths)
+  {
+    for (const ToolPath& tool_path : fragment)
+    {
+      for (const ToolPathSegment& segment : tool_path)
+      {
+        for (const Eigen::Isometry3d& w : segment)
+        {
+          // Add the point to the polyline
+          const int id = points->InsertNextPoint(w.translation().data());
+        }
+      }
+    }
+  }
+
+  // Set the number of ids for the polyline
+  polyLine->GetPointIds()->SetNumberOfIds(points->GetNumberOfPoints());
+
+  // Set the ids for the polyline
+  for (vtkIdType i = 0; i < points->GetNumberOfPoints(); i++)
+  {
+    polyLine->GetPointIds()->SetId(i, i);
+  }
```

Comment:
If we add lines between all parts of the vector of `ToolPaths`, the visualization gets a little cluttered and confusing, especially when there are mutliple regions and tool path approach/departure modifiers

![Screenshot_2024-05-01_15-08-44](https://github.com/ros-industrial/noether/assets/18411310/ce130918-84e0-49ad-9f02-fb59505df554)

![Screenshot_2024-05-01_15-09-10](https://github.com/ros-industrial/noether/assets/18411310/2e60604d-84f7-49d6-80a8-68a89551045e)

Ideally I think we should show:
- solid line between waypoints in a `ToolPathSegment`
- dashed line in between segments in a `ToolPath`
- dotted line in between tool path fragments in a `ToolPaths`
- no line between `ToolPaths` in a `std::vector<ToolPaths>`

I'm not sure what level of effort this would require, but I would at least like to start by only drawing solid lines between waypoints/segments in the same `ToolPath`, with no higher level connections. Could you add that change?

Reply:
Hi marip, thats a really good point. It was now easier to me to use colours instead of solid / dashed and dotted, because dotted seems only to show the dots at the added points positions and dashed also is not that nice to implement. I tried the easiest way possible, hope you like it.